### PR TITLE
decoder for ppar data block

### PR DESCRIPTION
**Pull request by yoctoyotta1024**
I made a first attempt at adding a decoder for a PPAR data block but I'm not sure it's correct. Please may you take a look?

Also I added a way we could then fead a decoded ppar block (as a dictionary / dataset) into decode_iq, but I don't understand what the manual says about how nfft is got from srt so for now I just kept nfft = 256 hardcoded... any ideas?

### CODE REVIEW

**Comment by d70-t** [comment]
I just had a very quick look for now.
* I'd remove the `config` dimension. Those variables should be 0-dimensional, because they shouldn't change. Probably we eventually want them (or some of them) to be attributes, but for now, I'd stay with 0-dim variables.
* The `decoders` dict probably wants to stay in the decoders module.

The manual sometimes isn't quite complete and some things sometimes are a bit different. It may be, that some settings still need to be hardcoded, or may have to be dependent only on the radar system instance key (the main block tag)...

**Comment by yoctoyotta1024** [comment]
Thanks Tobi, I didn't know you could have 0-D variables in a dataset! Did I get the syntax right? I tried to make the chanegd you suggested and I fixed a bug too.

**Comment by yoctoyotta1024** [comment]
Actually though I'm not sure about putting decoders dictionary into decoders.py because of two things. Firstly, how would you then use a ppar dict/dataset with it? I suppoer we could wap it in a function? Secondly I thought there might be cases when you want to select only some decoders and not have the full dictionary, then you could make your own dictionary but then I think that defeats the point of having one in decoders.py so maybe we don't want that?

**Comment by yoctoyotta1024** [comment]
Okay the ppar decoder is now working correctly.
I've created a way to use the ppar in the decoders of the data. It's a bit silly because at the moment a seperate dataset is made for PPAR before the dataset for the data, but that's just because read_pds is only for one file and I wanted to keep the dataset returned unchanged. When we process a whole flight (see serde branch for WIP) we can do this differently :)

Let me know you're thoughts!

File: hamp_radar/decoders.py [hunk 1]
```diff
@@ -1,4 +1,137 @@
...
+        b"FFTD": decode_iq(ppar),
+    }
+
+
+def decode_ppar(rawdata):
+    rawdata = rawdata[0]  # PPAR subblock is only 1 "frame"
+    return {
+        "prf": (
+            (),
+            rawdata[0:4].view("<i4"),
```
Comment (d70-t) [inline hamp_radar/decoders.py:28]: Did this one work out? I'd have thought this would stay a 1d array?

**Comment by d70-t** [comment]
Otherwise it's awesome and ready to merge 👍

**Pull request merged**